### PR TITLE
Allow SO_REUSEPORT

### PR DESCRIPTION
**Pull request by breerly**
@blampe @kriskowal @junchaowu @abhinav

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,7 +4,9 @@ Changes by Version
 0.22.4 (unreleased)
 -------------------
 
-- Nothing changed yet.
+- Added ``ruese_port`` param to ``TChannel`` which takes advantage
```
Comment (abhinav) [inline CHANGES.rst]: reuse_port

File: CHANGES.rst [hunk 2]
```diff
@@ -4,7 +4,9 @@ Changes by Version
...
+  of ``SO_REUSEPORT`` sockopt so that multiple processes can
```
Comment (abhinav) [inline CHANGES.rst]: > of the ``SO_REUSEPORT`` socket option [..]

**Comment by abhinav** [comment]
:+1:

File: .travis.yml [hunk 3]
```diff
@@ -1,11 +1,7 @@
...
-
-#
-# use container-based infrastructure
-# @see http://docs.travis-ci.com/user/workers/container-based-infrastructure/
-# @see http://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/
-#
-sudo: false
+sudo: required
+services:
+    - docker
```
Comment (breerly) [inline .travis.yml]: somehow these two things are required for `SO_REUSEPORT` to not blowup on travis :\

**Comment by abhinav** [comment]
:shipit:

File: CHANGES.rst (hunk 1 above)
Comment (breerly) [inline CHANGES.rst]: :P

File: CHANGES.rst (hunk 2 above)
Comment (breerly) [inline CHANGES.rst]: :P

**Pull request merged**